packages/interface-unobtrusive-control-ui/src/index.tsx

```diff
@@ -0,0 +1,209 @@
+
+  const [needsIntroduction, setNeedsIntroduction] = useState(
+    hasPendingDescisions
+  )
+  const [slideUp, setSlideUp] = useState(false)
+
+  const toggleControlForm = useCallback(
+    e => {
+      e.preventDefault()
+      setShowIntroduction(false)
+      setSlideUp(v => !v)
+    },
+    [setSlideUp, setShowIntroduction]
+  )
+
+  // @todo calling onSubmit causes rerender
+  const onSubmitCb = useCallback(
+    values => {
+      const enabled = []
+      for (const [key, value] of Object.entries(values)) {
+        if (value) {
+          enabled.push(key)
+        }
+      }
+      setSlideUp(false)
+      onSubmit({ enabled })
+    },
+    [onSubmit]
+  )
+
+  const initial = useMemo(() => {
+    const initialState: FormState = {}
+    for (const integration of integrations) {
+      initialState[integration.id] = initialValues.enabled.includes(
+        integration.id
+      )
+    }
+
+    return initialState
+  }, [integrations, initialValues])
+
+  // Check if component was mounted for SSR
+  const [isMounted, setIsMounted] = useState(false)
+  useEffect(() => setIsMounted(true), [setIsMounted])
+
+  // Introduction animation
+  useEffect(() => {
+    if (!isMounted || !needsIntroduction) {
+      return
+    }
+    setShowIntroduction(true)
+    setNeedsIntroduction(false)
+
+    // Wait for no user interaction
+    // @todo actually check it
+    window.setTimeout(() => {
+      setSlideUp(true)
+
+      // Keep it visible to the user
+      window.setTimeout(() => {
+        setSlideUp(false)
+
+        // Wait with swapping contentn till slide down is finished
+        window.setTimeout(() => {
+          setShowIntroduction(false)
+        }, introductionTransitionDuration)
+      }, introductionVisibleDuration)
+    }, introductionNoActionDelay)
+  }, [
+    isMounted,
+    setShowIntroduction,
+    needsIntroduction,
+    introductionNoActionDelay,
+    introductionVisibleDuration,
+    introductionTransitionDuration,
+  ])
+
+  // Do not render the interface on SSR.
+  if (!isMounted) {
+    return null
+  }
+
+  return (
+    <div
+      className={clsx(
+        styles.wrapper,
+        styles.pane,
+        styles.slide,
+        slideUp && styles.slideUp
+      )}
+      style={{
+        transitionDuration: `${introductionTransitionDuration}ms`,
+      }}
+      id="consent-control-ui"
+    >
+      <button
+        className={clsx(styles.toggleButton)}
+        title={`Toggle website settings visibility`}
+        onClick={toggleControlForm}
+      >
+        <div className={clsx(styles.pane, styles.toggleButtonContent)}>
+          <ToggleIcon className={clsx(slideUp && styles.inverted)} />
+        </div>
+      </button>
+      {showIntroduction ? (
+        <div className={clsx(styles.introduction)}>
+          Some features got disabled in respect of your privacy.
+          <br /> Click here to learn more!
+        </div>
+      ) : (
+        <Form
```

Comment:
Consider to have the actual form in its own component as well, if we do that, we might to rename `<ConsentForm />` to `<ConsentUserInterface/>`